### PR TITLE
Add buffer pool watermark support in watermarkstat and counterpoll

### PR DESCRIPTION
**Pull request by wendani**
$ watermarkstat -t buffer_pool -p
Shared pool maximum occupancy:
                 Pool    Bytes
---------------------  -------
 egress_lossless_pool   570336
    egress_lossy_pool  2054624
ingress_lossless_pool     1664

$ counterpoll watermark enable

$ counterpoll watermark interval 20000

Apr 23 17:49:44.686196 
Apr 23 17:50:04.686427 
Apr 23 17:50:24.686704 

$ counterpoll watermark disable

$ counterpoll show
Type                        Interval (in ms)    Status
--------------------------  ------------------  --------
QUEUE_STAT                  default (10000)     enable
PORT_STAT                   default (1000)      enable
QUEUE_WATERMARK_STAT        20000               disable
PG_WATERMARK_STAT           20000               disable
BUFFER_POOL_WATERMARK_STAT  20000               disable

**- What I did**

**- How I did it**

**- How to verify it**
tested on brcm dut
**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**

-->

### CODE REVIEW

**Comment by wendani** [comment]
@mykolaf for review

File: counterpoll/main.py [hunk 1]
```diff
@@ -172,6 +180,8 @@ def show():
         data.append(["QUEUE_WATERMARK_STAT", queue_wm_info["POLL_INTERVAL"] if 'POLL_INTERVAL' in queue_wm_info else 'default (10000)', queue_wm_info["FLEX_COUNTER_STATUS"] if 'FLEX_COUNTER_STATUS' in queue_wm_info else 'disable' ])
     if pg_wm_info:
         data.append(["PG_WATERMARK_STAT", pg_wm_info["POLL_INTERVAL"] if 'POLL_INTERVAL' in pg_wm_info else 'default (10000)', pg_wm_info["FLEX_COUNTER_STATUS"] if 'FLEX_COUNTER_STATUS' in pg_wm_info else 'disable'])
+    if buffer_pool_wm_info:
+        data.append(["BUFFER_POOL_WATERMARK_STAT", buffer_pool_wm_info["POLL_INTERVAL"] if 'POLL_INTERVAL' in buffer_pool_wm_info else 'default (10000)', buffer_pool_wm_info["FLEX_COUNTER_STATUS"] if 'FLEX_COUNTER_STATUS' in buffer_pool_wm_info else 'disable'])
```
Comment (qiluo-msft) [inline counterpoll/main.py]: >buffer_pool_wm_info["FLEX_COUNTER_STATUS"] if 'FLEX_COUNTER_STATUS' in buffer_pool_wm_info else 'disable'] [](start = 154, length = 106)

You could use dict.get() function and provide a default value. These long lines are not so goo.

File: scripts/watermarkstat [hunk 2]
```diff
@@ -127,27 +129,36 @@ class Watermarkstat(object):
             port = self.port_name_map[get_pg_port(counter_pg_name_map[pg])]
             self.port_pg_map[port][pg] = counter_pg_name_map[pg]
 
+        # Get all buffer pools
+        self.buffer_pool_name_to_oid_map = self.counters_db.get_all(self.counters_db.COUNTERS_DB, COUNTERS_BUFFER_POOL_NAME_MAP)
+        if self.buffer_pool_name_to_oid_map is None:
+            print "COUNTERS_BUFFER_POOL_NAME_MAP is empty!"
```
Comment (qiluo-msft) [inline scripts/watermarkstat]: >print [](start = 12, length = 5)

print to stderr

File: scripts/watermarkstat [hunk 3]
```diff
@@ -184,14 +195,25 @@ class Watermarkstat(object):
         cntr = tuple(fields)
         return cntr
 
-    def get_print_all_stat(self, table_prefix, type):
-        # Get stat for each port
+    def get_print_all_stat(self, table_prefix, key):
```
Comment (qiluo-msft) [inline scripts/watermarkstat]: >get_print_all_stat [](start = 8, length = 18)

print_all_stat

File: scripts/watermarkstat [hunk 4]
```diff
@@ -214,12 +236,15 @@ Examples:
   watermarkstat -t q_shared_multi -c
   watermarkstat -p -t pg_shared
   watermarkstat -p -t q_shared_multi -c
+  watermarkstat -t buffer_pool
```
Comment (mykolaf) [inline scripts/watermarkstat]: I don't see any changes in show/main.py.
Are there "show command" alias for this? Is the CLI command format decided?

**Comment by wendani** [comment]
retest this please

**Comment by wendani** [comment]
retest this please

**Comment by stcheng** [comment]
retest this please

**Pull request merged**